File: Dagger2TestAndroid/app/src/main/java/com/example/raghu/dagger2testandroid/dagger/AppComponent.java
package com.example.raghu.dagger2testandroid.dagger;

import android.app.Application;

import com.example.raghu.dagger2testandroid.DemoApplication;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

/**
 * Created by raghu on 4/8/17.
 */
@Singleton
@Component(modules = {
        AndroidInjectionModule.class,
        AppModule.class,
        ActivityBuilder.class})
public interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance Builder application(Application application);
        AppComponent build();
    }

    void inject(DemoApplication app);
}

File: SpecBeeMapAssignment/app/src/main/java/com/example/raghu/specbeemapassignment/dagger/AppComponent.kt
package com.example.raghu.dagger2testandroid.dagger

import android.app.Application

import com.example.raghu.specbeemapassignment.DemoApplication

import javax.inject.Singleton

import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule

/**
 * Created by raghu on 4/8/17.
 */
@Singleton
@Component(modules = arrayOf(AndroidInjectionModule::class, AppModule::class, ActivityBuilder::class))
interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance fun application(application: Application): Builder
        fun build(): AppComponent
    }

    fun inject(app: DemoApplication)
}
